src/lib/block_if.c

```diff
@@ -213,6 +224,10 @@ block_delete(struct blockif_ctxt *bc, off_t offset, off_t len)
 	int ret = -1;
 #ifdef __FreeBSD__
 	off_t arg[2] = { offset, len };
+#elif __APPLE__
+	struct fpunchhole arg = { .fp_flags = 0, .reserved = 0, .fp_offset = offset, .fp_length = len };
+	size_t len_to_zero;
+	ssize_t aligned_offset, aligned_length, written;
```

Comment:
Of these three I think the two `aligned_*` can be `size_t`. If there is any chance of underflow then that ought to be checked before the calculation, and I think because both come from `ALIGN*` there is no such possibility.

Reply:
Since we are using c99 I think `len_to_zero` and `written` can both be defined at the point of use (in both cases twice, in two disconnected scopes). e.g. you can write:
```
size_t len_to_zero = aligned_offset - arg.fp_offset;
...
ssize_t written = pwrite(bc->bc_fd, bc->bc_delete_zero_buf, len_to_zero, arg.fp_offset);
```
for the trim at the start.